internal/components/base/reconcile.go

```diff
@@ -0,0 +1,123 @@
+/*
```

Comment:
I'm wondering if it would make sense to have all these files in `pkg` instead of `internal`. I guess it would make sense at some point to reuse some of these functions/constants whatever externally, but I'm not sure, rather asking whether `internal` was a deliberate decision or we should at least consider `pkg`.

Reply:
I can't really see reusing the reconciliation logic as is, so I'd say leave it in `internal` for now. Once we think that it should be reused, we should definitely move it.